pkg/nodeconfig/nodeconfig.go

```diff
@@ -534,6 +540,22 @@ func (nc *nodeConfig) UpdateKubeletClientCA(contents []byte) error {
 	return nil
 }
 
+// ensureTrustedCABundle gets the trsuted CA ConfigMap and ensures the cert bundle on the instance has up-to-date data
+func (nc *nodeConfig) ensureTrustedCABundle() error {
+	trustedCA, err := nc.k8sclientset.CoreV1().ConfigMaps(nc.wmcoNamespace).Get(context.TODO(),
+		certificates.ProxyCertsConfigMap, meta.GetOptions{})
```

Comment:
Please use the cached client here. applies to all calls which deal with objects we watch.

Reply:
Perhaps a check should be made to ensure that the CM has the key `certificates.CABundleKey`?
Otherwise that could cause issues within `UpdateTrustedCABundleFile`

Reply:
> Perhaps a check should be made to ensure that the CM has the key certificates.CABundleKey?
> Otherwise that could cause issues within UpdateTrustedCABundleFile

I didn't add this check because I was thinking that if the key doesn't exist, it could mean a user cleared out their settings/removed all their additional certs. In which case UpdateTrustedCABundleFile(), will reflect this by setting the file contents as empty (and then WICD can react to that)